### PR TITLE
Ban a member

### PR DESCRIPTION
**Pull request by Vamshi99**
Add a command to remove a member from newcomer team,
blacklist member from using corobo. Limit usage of
this command to mainatiners and not allowed in
private chat

Closes https://github.com/coala/corobo/issues/462

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

**Comment by jayvdb** [comment]
This has way too much in it, and is going in the opposite direction to https://github.com/coala/corobo/issues/420 .  Hopefully https://github.com/coala/corobo/issues/462#issuecomment-349526147 will give us some well tested implementations in IGitt which can be easily used here.

**Comment by jayvdb** [comment]
@gitmate-bot rebase

File: plugins/labhub.py [hunk 1]
```diff
@@ -381,3 +382,61 @@ def pr_stats(self, msg, match):
...
+                           'Accept': 'application/json'}
+                for room in self.bot_config.ROOMS_TO_JOIN:
+                    data = json.dumps({"username": unban_user})
+                    r = requests.delete('https://api.gitter.im/v1/rooms/'
+                                        '{}/bans'.format(room.idd),
+                                        headers=headers,
+                                        data=data)
+                    if r.status_code != 200:
+                        return("Error while unbanning user from"
+                               "{} gitter channel".format(room.uri))
```
Comment (meetmangukiya) [inline plugins/labhub.py]: Use `yield` so it can continue to unban from other rooms. Same for ban.

File: tests/labhub_test.py [hunk 2]
```diff
@@ -12,6 +12,8 @@
 from IGitt.GitHub.GitHubIssue import GitHubIssue
 
 from errbot.backends.test import TestBot
+from errbot.backends.base import Message
+from errbot import BotPlugin
```
Comment (meetmangukiya) [inline tests/labhub_test.py]: Unused import

File: tests/labhub_test.py [hunk 3]
```diff
@@ -12,6 +12,8 @@
 from IGitt.GitHub.GitHubIssue import GitHubIssue
 
 from errbot.backends.test import TestBot
+from errbot.backends.base import Message
```
Comment (meetmangukiya) [inline tests/labhub_test.py]: Unused import

File: utils/filters.py [hunk 4]
```diff
@@ -21,6 +21,7 @@ def filters(self, msg, cmd, args, dry_run):
             if cmd in commands and msg.frm.room.uri == room:
                 return None, None, None
         return msg, cmd, args
+
```
Comment (meetmangukiya) [inline utils/filters.py]: unnecessary change

File: tests/labhub_test.py [hunk 5]
```diff
@@ -346,3 +348,50 @@ def test_invite_me(self):
...
+        testbot.assertCommand("!ban @testuser","you\'re banned")
+        # Already banned username
+        testbot.bot_config.USERS_TO_BLACKLIST.append("testuser")
+        testbot.assertCommand("!ban @testuser","user is already")
+        testbot.bot_config.USERS_TO_BLACKLIST.remove("testuser")
+        # Command used by a non-maintainer
+        mock_maint_team.is_member.return_value = False
+        testbot.assertCommand("!ban @testuser","only maintainers")
+     
+    def test_ban(self):
```
Comment (meetmangukiya) [inline tests/labhub_test.py]: `test_unban`*

**Comment by Vamshi99** [comment]
@meetmangukiya I can't work on this right now. Can you unassign me on this and close the PR? @nvzard can use this code in his PR.

File: plugins/labhub.py [hunk 6]
```diff
@@ -381,3 +382,61 @@ def pr_stats(self, msg, match):
...
+            self.bot_config.USERS_TO_BLACKLIST.append(ban_user)
+            gitter_token = self.bot_config.BOT_IDENTITY['token']
+            headers = {'Authorization': 'Bearer '
+                       '{}'.format(gitter_token),
+                       'Content-Type': 'application/json',
+                       'Accept': 'application/json'}
+            for room in self.bot_config.ROOMS_TO_JOIN:
+                data = json.dumps({"username": ban_user})
+                r = requests.post('https://api.gitter.im/v1/rooms/'
+                                  '{}/bans'.format(room.idd),
```
Comment (nvzard) [inline plugins/labhub.py]: `room` is string type and it has no attribute 'idd'

**Comment by nvzard** [comment]
I'll not be using this code since I already have a working tested implemented ready. https://github.com/coala/corobo/pull/553
